doc/reference/configuration/configuration_reference.rst

```diff
@@ -2877,6 +2878,120 @@ This section describes configuration parameters related to roles.
 
     **Since:** :doc:`3.0.0 </release/3.0.0>`.
 
+    Specify the roles of an instance.
+    To specify a role's configuration, use the :ref:`roles_cfg <configuration_reference_roles_cfg>` option.
+
+    There are two types of application roles:
+
+    -   Built-in roles. For example, the `CRUD <https://github.com/tarantool/crud>`__ module provides the ``roles.crud-storage`` and ``roles.crud-router`` roles that enable CRUD operations in a sharded cluster.
+    -   Custom roles. For example, you create a custom role to define a stored procedure or implement a supplementary service, such as an e-mail notifier or a replicator.
+
+    To create a custom role, you need to define the following methods and fields:
+
+    -   ``validate()``
```

Comment:
The idea of this function is to provide a validation feedback earlier, so all the `validate` functions for all the enabled roles are executed first and only after that the `apply()` functions are executed. I think that it worth to mention it, because otherwise it is not clear why not to just call some `myvalidate()` directly from `apply()`.

Reply:
Extended the description of the `apply` function:

```
    -   ``apply()``

        Apply a role's configuration.
        ``apply()`` is called after ``validate()`` is executed for all the enabled roles.
        As the ``validate()`` function, ``apply()`` is called on instance startup or when the configuration is reloaded for the instance with this role.
```